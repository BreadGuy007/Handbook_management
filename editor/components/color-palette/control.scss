.editor-color-palette-control__color-palette {
	margin-top: .6rem;
	margin-bottom: 1.4rem;
}
