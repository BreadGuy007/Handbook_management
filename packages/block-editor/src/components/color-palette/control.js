/**
 * WordPress dependencies
 */
import { BaseControl, ColorIndicator } from '@wordpress/components';
import { ifCondition, compose } from '@wordpress/compose';
import { Fragment } from '@wordpress/element';
import { sprintf, __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import ColorPalette from './';
import withColorContext from './with-color-context';
import { getColorObjectByColorValue } from '../colors';

// translators: first %s: The type of color (e.g. background color), second %s: the color name or value (e.g. red or #ff0000)
const colorIndicatorAriaLabel = __( '(current %s: %s)' );

export function ColorPaletteControl( {
	colors,
	disableCustomColors,
	label,
	onChange,
	value,
} ) {
	const colorObject = getColorObjectByColorValue( colors, value );
	const colorName = colorObject && colorObject.name;
	const ariaLabel = sprintf( colorIndicatorAriaLabel, label.toLowerCase(), colorName || value );

	const labelElement = (
		<Fragment>
			{ label }
			{ value && (
				<ColorIndicator
					colorValue={ value }
					aria-label={ ariaLabel }
				/>
			) }
		</Fragment>
	);

	return (
		<BaseControl
			className="editor-color-palette-control block-editor-color-palette-control"
			label={ labelElement }>
			<ColorPalette
				className="editor-color-palette-control__color-palette block-editor-color-palette-control__color-palette"
				value={ value }
				onChange={ onChange }
				{ ... { colors, disableCustomColors } }
			/>
		</BaseControl>
	);
}

export default compose( [
	withColorContext,
	ifCondition( ( { hasColorsToChoose } ) => hasColorsToChoose ),
] )( ColorPaletteControl );
