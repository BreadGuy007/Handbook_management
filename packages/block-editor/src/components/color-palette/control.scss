.block-editor-color-palette-control__color-palette {
	display: inline-block;
	margin-top: 0.6rem;
}
