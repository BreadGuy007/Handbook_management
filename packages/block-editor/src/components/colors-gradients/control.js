/**
 * External dependencies
 */
import classnames from 'classnames';
import { every, isEmpty, pick } from 'lodash';

/**
 * WordPress dependencies
 */
import { useState } from '@wordpress/element';
import {
	BaseControl,
	Button,
	ButtonGroup,
	ColorIndicator,
	ColorPalette,
	__experimentalGradientPicker as GradientPicker,
} from '@wordpress/components';
import { sprintf, __ } from '@wordpress/i18n';
import { useSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import { getColorObjectByColorValue } from '../colors';
import { __experimentalGetGradientObjectByGradientValue } from '../gradients';

// translators: first %s: the color name or value (e.g. red or #ff0000)
const colorIndicatorAriaLabel = __( '(Color: %s)' );

// translators: first %s: the gradient name or value (e.g. red to green or linear-gradient(135deg,rgba(6,147,227,1) 0%,rgb(155,81,224) 100%)
const gradientIndicatorAriaLabel = __( '(Gradient: %s)' );

const colorsAndGradientKeys = [
	'colors',
	'disableCustomColors',
	'gradients',
	'disableCustomGradients',
];

function VisualLabel( {
	colors,
	gradients,
	label,
	currentTab,
	colorValue,
	gradientValue,
} ) {
	let value, ariaLabel;
	if ( currentTab === 'color' ) {
		if ( colorValue ) {
			value = colorValue;
			const colorObject = getColorObjectByColorValue( colors, value );
			const colorName = colorObject && colorObject.name;
			ariaLabel = sprintf( colorIndicatorAriaLabel, colorName || value );
		}
	} else if ( currentTab === 'gradient' && gradientValue ) {
		value = gradientValue;
		const gradientObject = __experimentalGetGradientObjectByGradientValue(
			gradients,
			value
		);
		const gradientName = gradientObject && gradientObject.name;
		ariaLabel = sprintf(
			gradientIndicatorAriaLabel,
			gradientName || value
		);
	}

	return (
		<>
			{ label }
			{ !! value && (
				<ColorIndicator colorValue={ value } aria-label={ ariaLabel } />
			) }
		</>
	);
}

function ColorGradientControlInner( {
	colors,
	gradients,
	disableCustomColors,
	disableCustomGradients,
	className,
	label,
	onColorChange,
	onGradientChange,
	colorValue,
	gradientValue,
} ) {
	const canChooseAColor =
		onColorChange && ( ! isEmpty( colors ) || ! disableCustomColors );
	const canChooseAGradient =
		onGradientChange &&
		( ! isEmpty( gradients ) || ! disableCustomGradients );
	const [ currentTab, setCurrentTab ] = useState(
		gradientValue ? 'gradient' : !! canChooseAColor && 'color'
	);

	if ( ! canChooseAColor && ! canChooseAGradient ) {
		return null;
	}
	return (
		<BaseControl
			className={ classnames(
				'block-editor-color-gradient-control',
				className
			) }
		>
			<fieldset>
				<legend>
					<div className="block-editor-color-gradient-control__color-indicator">
						<BaseControl.VisualLabel>
							<VisualLabel
								currentTab={ currentTab }
								label={ label }
								colorValue={ colorValue }
								gradientValue={ gradientValue }
							/>
						</BaseControl.VisualLabel>
					</div>
				</legend>
				{ canChooseAColor && canChooseAGradient && (
					<ButtonGroup className="block-editor-color-gradient-control__button-tabs">
						<Button
							isSmall
							isPrimary={ currentTab === 'color' }
							isSecondary={ currentTab !== 'color' }
							onClick={ () => setCurrentTab( 'color' ) }
						>
							{ __( 'Solid' ) }
						</Button>
						<Button
							isSmall
							isPrimary={ currentTab === 'gradient' }
							isSecondary={ currentTab !== 'gradient' }
							onClick={ () => setCurrentTab( 'gradient' ) }
						>
							{ __( 'Gradient' ) }
						</Button>
					</ButtonGroup>
				) }
				{ ( currentTab === 'color' || ! canChooseAGradient ) && (
					<ColorPalette
						value={ colorValue }
						onChange={
							canChooseAGradient
								? ( newColor ) => {
										onColorChange( newColor );
										onGradientChange();
								  }
								: onColorChange
						}
						{ ...{ colors, disableCustomColors } }
					/>
				) }
				{ ( currentTab === 'gradient' || ! canChooseAColor ) && (
					<GradientPicker
						value={ gradientValue }
						onChange={
							canChooseAColor
								? ( newGradient ) => {
										onGradientChange( newGradient );
										onColorChange();
								  }
								: onGradientChange
						}
						{ ...{ gradients, disableCustomGradients } }
					/>
				) }
			</fieldset>
		</BaseControl>
	);
}

function ColorGradientControlSelect( props ) {
	const colorGradientSettings = useSelect( ( select ) => {
		const settings = select( 'core/block-editor' ).getSettings();
		return pick( settings, colorsAndGradientKeys );
	} );
	return (
		<ColorGradientControlInner
			{ ...{ ...colorGradientSettings, ...props } }
		/>
	);
}

function ColorGradientControl( props ) {
	if (
		every( colorsAndGradientKeys, ( key ) => props.hasOwnProperty( key ) )
	) {
		return <ColorGradientControlInner { ...props } />;
	}
	return <ColorGradientControlSelect { ...props } />;
}

export default ColorGradientControl;
