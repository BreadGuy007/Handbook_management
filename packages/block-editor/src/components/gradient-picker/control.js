
/**
 * External dependencies
 */
import classnames from 'classnames';
import { isEmpty, pick } from 'lodash';

/**
 * WordPress dependencies
 */
import { BaseControl } from '@wordpress/components';
import { __ } from '@wordpress/i18n';
import { useSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import GradientPicker from './';

export default function( { className, value, onChange, label = __( 'Gradient Presets' ), ...props } ) {
	const { gradients = [], disableCustomGradients } = useSelect( ( select ) => (
		pick(
			select( 'core/block-editor' ).getSettings(),
			[ 'gradients', 'disableCustomGradients' ]
		)
	) );
	if ( isEmpty( gradients ) && disableCustomGradients ) {
		return null;
	}
	return (
		<BaseControl
			className={ classnames(
				'block-editor-gradient-picker-control',
				className
			) }
		>
			<BaseControl.VisualLabel>
				{ label }
			</BaseControl.VisualLabel>
			<GradientPicker
				value={ value }
				onChange={ onChange }
				className="block-editor-gradient-picker-control__gradient-picker-presets"
				gradients={ gradients }
				disableCustomGradients={ disableCustomGradients }
				{ ...props }
			/>
		</BaseControl>
	);
}
