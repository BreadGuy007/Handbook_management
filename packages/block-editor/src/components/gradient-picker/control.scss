.block-editor-gradient-picker-control__gradient-picker-presets {
	display: inline-block;
	margin-top: 0.6rem;
}
