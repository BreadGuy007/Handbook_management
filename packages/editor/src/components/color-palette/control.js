/**
 * WordPress dependencies
 */
import { BaseControl, ColorIndicator } from '@wordpress/components';
import { Fragment } from '@wordpress/element';
import { sprintf, __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import ColorPalette from './';
import withColorContext from './with-color-context';
import { getColorName } from '../colors';

// translators: first %s: The type of color (e.g. background color), second %s: the color name or value (e.g. red or #ff0000)
const colorIndicatorAriaLabel = __( '(current %s: %s)' );

export function ColorPaletteControl( { label, value, onChange, colors } ) {
	const colorName = getColorName( colors, value );
	const ariaLabel = sprintf( colorIndicatorAriaLabel, label.toLowerCase(), colorName || value );

	const labelElement = (
		<Fragment>
			{ label }
			{ value && (
				<ColorIndicator
					colorValue={ value }
					aria-label={ ariaLabel }
				/>
			) }
		</Fragment>
	);

	return (
		<BaseControl
			className="editor-color-palette-control"
			label={ labelElement }>
			<ColorPalette
				className="editor-color-palette-control__color-palette"
				value={ value }
				onChange={ onChange }
			/>
		</BaseControl>
	);
}

export default withColorContext( ColorPaletteControl );
