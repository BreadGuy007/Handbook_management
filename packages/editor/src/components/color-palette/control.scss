.editor-color-palette-control__color-palette {
	margin-top: 0.6rem;
	margin-bottom: 1.4rem;
}
