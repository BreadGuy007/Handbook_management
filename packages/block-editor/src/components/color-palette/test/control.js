/**
 * External dependencies
 */
import { shallow } from 'enzyme';
import { noop } from 'lodash';

/**
 * Internal dependencies
 */
import { ColorPaletteControl } from '../control';

describe( 'ColorPaletteControl', () => {
	it( 'matches the snapshot', () => {
		const wrapper = shallow(
			<ColorPaletteControl
				label="Test Color"
				value="#f00"
				colors={ [ { color: '#f00', name: 'red' } ] }
				onChange={ noop }
			/>
		);

		expect( wrapper ).toMatchSnapshot();
	} );
} );
