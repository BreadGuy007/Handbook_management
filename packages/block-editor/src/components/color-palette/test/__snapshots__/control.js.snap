// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPaletteControl matches the snapshot 1`] = `
<div
  className="components-base-control block-editor-color-gradient-control"
>
  <div
    className="components-base-control__field"
  >
    <fieldset>
      <legend>
        <div
          className="block-editor-color-gradient-control__color-indicator"
        >
          <span
            className="components-base-control__label"
          >
            Test Color
            <span
              aria-label="(Color: #f00)"
              className="component-color-indicator"
              style={
                Object {
                  "background": "#f00",
                }
              }
            />
          </span>
        </div>
      </legend>
      <div
        className="components-circular-option-picker"
      >
        <div
          className="components-circular-option-picker__option-wrapper"
        >
          <button
            aria-label="Color: red"
            aria-pressed={true}
            className="components-button components-circular-option-picker__option is-pressed"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "#f00",
                "color": "#f00",
              }
            }
            type="button"
          />
          <svg
            aria-hidden="true"
            fill="#000000"
            focusable="false"
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 18.6L3.5 13l1-1L9 16.4l9.5-9.9 1 1z"
            />
          </svg>
        </div>
        <div
          className="components-circular-option-picker__custom-clear-wrapper"
        >
          <div
            className="components-dropdown components-circular-option-picker__dropdown-link-action"
          >
            <button
              aria-expanded={false}
              aria-label="Custom color picker"
              className="components-button is-link"
              onClick={[Function]}
              type="button"
            >
              Custom color
            </button>
          </div>
          <button
            className="components-button components-circular-option-picker__clear is-secondary is-small"
            onClick={[Function]}
            type="button"
          >
            Clear
          </button>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;
